Cloud Gateways: Custom DNS management (#7163)
Kong/docs.konghq.com: app/konnect/reference/custom-dns.md

---
title: Custom Domains
---

{{site.konnect_short_name}} integrates domain name management and configuration in the Dev Portal and with [managed data planes](/konnect/dedicated-cloud-gateways/). 

> ## Managed Data Planes

Reviewer: ```suggestion
## Managed data planes
```

### {{site.konnect_short_name}} configuration

1. Open {% konnect_icon runtimes %} **Gateway Manager**, choose a control plane to open the **Overview** dashboard, then click **Connect**.
    

>     The **Connect** menu will open and display the URL for the **Public Edge DNS**, save this URL.

Reviewer: ```suggestion
    The **Connect** menu will open and display the URL for the **Public Edge DNS**. Save this URL.
```

1. Select **Custom Domains** from the side navigation, then **New Custom Domain**, and enter your domain name.

    Save the values that appear under **CNAME** and **Content**. 

> ### Domain Register configuration

Reviewer: ```suggestion
### Domain registrar configuration
```
I think that's right, feel free to ignore if I'm wrong here.

1. Log in to your domain registrar's dashboard.
1. Navigate to the DNS settings section. This area might be labeled differently depending on your registrar.

> 1. Locate the option to add a new CNAME record and create the following records with the values saved in the previous step. For example in AWS Route 53, it would look like this: 

Reviewer: ```suggestion
1. Locate the option to add a new CNAME record and create the following records using the values saved in the [{{site.konnect_short_name}} configuration](#konnect-configuration) section. For example, in AWS Route 53, it would look like this: 
```

| Host Name                       | Record Type | Routing Policy | Alias | Evaluate Target Health | Value                                                | TTL |
|---------------------------------|-------------|----------------|-------|------------------------|------------------------------------------------------|-----|
| `_acme-challenge.example.com` | CNAME       | Simple         | No    | No                     | `_acme-challenge.9e454bcfec.acme.gateways.konghq.com`| 300 |
| `example.com`             | CNAME       | Simple         | No    | No                     | `9e454bcfec.gateways.konghq.com`                     | 300 |

## Dev Portal 

Every Dev Portal instance has an auto-generated default URL. You can also manage custom URLs within {{site.konnect_short_name}}. This gives users the ability to access the Dev Portal from either the default URL, for example `https://example.us.portal.konghq.com`, or a custom URL like `portal.example.com`.

To add a custom URL to Dev Portal, you need:

> * A domain and access to configure the domain's DNS `CNAME` records.
> * Your organization's auto-generated default Dev Portal URL.
> * A [CAA DNS](https://datatracker.ietf.org/doc/html/rfc6844) record that allows `pki.goog` only if any pre-existing CAA DNS records are present on the domain.

Reviewer 2: ```suggestion
* A domain and access to configure the domain's DNS `CNAME` records
* Your organization's auto-generated default Dev Portal URL
* A [CAA DNS](https://datatracker.ietf.org/doc/html/rfc6844) record that only allows `pki.goog` if any pre-existing CAA DNS records are present on the domain
```

* TLDs restricted by US export laws:
  * `.af` Afghanistan
  * `.by` The Republic of Belarus
  * `.cu` Cuba
  * `.er` Eritrea
  * `.gn` Guinea
  * `.ir` Islamic Republic of Iran
  * `.kp` Democratic People's Republic of Korea
  * `.lr` Liberia
  * `.ru` The Russian Federation
  * `.ss` South Sudan
  * `.su` Soviet Union
  * `.sy` Syrian Arab Republic
  * `.zw` Zimbabwe

If you have any questions, [contact Support](https://support.konghq.com).

### Delete a custom URL {#delete-url}

Delete a custom Dev Portal URL through your organization's {{site.konnect_short_name}} admin UI.

1. In {{site.konnect_short_name}}, open {% konnect_icon dev-portal %} **Dev Portal**, then click **Settings**.

2. Open the **Portal Domain** tab.

3. Click **Delete Custom Domain**

## Troubleshoot DNS {#troubleshoot}

After the DNS verification process is complete, {{site.konnect_short_name}} will attempt to automatically generate
an SSL certificate for your custom domain. This process can take several hours. If you attempt to access the custom URL from a browser _before_ the certificate generation process is finished,
you will receive an SSL certificate error.  If this process takes more than 24 hours,

> please check if the new DNS record is correctly applied. You can use the `dig` tool to troubleshoot your DNS:

Reviewer 2: ```suggestion
check if the new DNS record is correctly applied. You can use the `dig` tool to troubleshoot your DNS:
```
